components/scream/cime_config/buildnml

```diff
@@ -176,304 +278,106 @@ def evaluate_selectors(element, case, ez_selectors):
-    {   'test.yaml': {   'group1': {   'interleave': ['string', None],
-                                       'testval': ['integer', None],
-                                       'testval2': ['integer', None]},
-                         'group2': {   'bar': ['integer', None],
-                                       'bat': ['integer', None],
-                                       'baz': ['integer', None],
-                                       'foo': ['integer', None],
-                                       'interleave': ['string', None],
-                                       'prec_tester': ['integer', None],
-                                       'sel_regex_no_match': ['integer', None],
-                                       'testval': ['integer', None],
-                                       'testval2': ['integer', None]}}}
-    >>> ######################################################################
-    >>> test_metadata_override = '''
+    On input, xml contains the parsed content of namelist_defaults_scream.xml.
+    On output, it contains the input parameters for this case.
+
+    >>> from eamxx_buildnml_impl import MockCase
+    >>> case = MockCase({'foo':1, 'bar':2, 'baz_bar':'blah', 'baz2_bar2':4})
+    >>> xml = '''
     ... <namelist_defaults>
-    ...   <selectors>
-    ...     <selector name="hgrid" case_env="ATM_GRID"/>
-    ...   </selectors>
-    ...   <file name="test.yaml" format="yaml">
-    ...     <testval valid_values="1,2,3">1</testval>
-    ...     <testval hgrid="ne4ne4" valid_values="1,2,3,4">2</testval>
-    ...     <testval hgrid="no_match" valid_values="1,2,3,4,5">3</testval>
-    ...     <testval2 valid_values="1,2,3">1</testval2>
-    ...     <testval2 hgrid="ne4ne4">2</testval2>
-    ...     <testval2 hgrid="no_match">3</testval2>
-    ...   </file>
+    ...     <selectors/>
+    ...     <input_files/>
+    ...     <generated_files/>
+    ...     <atmosphere_processes_defaults>
+    ...         <atm_procs_list>(P1,P2)</atm_procs_list>
+    ...         <atm_proc_base>
+    ...             <prop1>zero</prop1>
+    ...         </atm_proc_base>
+    ...         <atm_proc_group inherit="atm_proc_base">
+    ...             <atm_procs_list>NONE</atm_procs_list>
+    ...             <prop2>one</prop2>
+    ...         </atm_proc_group>
+    ...         <P1 inherit="atm_proc_base">
+    ...             <prop1>two</prop1>
+    ...         </P1>
+    ...         <P2 inherit="atm_proc_base">
+    ...         </P2>
+    ...     </atmosphere_processes_defaults>
     ... </namelist_defaults>
     ... '''
-    >>> _create_raw_xml_file_impl(case, test_metadata_override)[1]
-    {'test.yaml': {'testval': ['integer', '1,2,3,4'], 'testval2': ['integer', '1,2,3']}}
+    >>> import xml.etree.ElementTree as ET
+    >>> defaults = ET.fromstring(xml)
+    >>> generated = _create_raw_xml_file_impl(case,defaults)
+    >>> aps = get_child(generated,"atmosphere_processes")
+    >>> for c in aps:
```

Comment:
Instead of probing objects in the doctests, why not just dump the raw xml and compare all of it at once?

Reply:
What do you mean with "dump"? I don't want doctest to create a file. I guess ordered_dump can dump to a dictionary?